### PR TITLE
TASK-38016: Enhance suggester service to allow restrict assigning to project participants

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
- To allow retrieving the list of participants of project , we add a specific url to get participants from task addons;

### CODE REVIEW

**Review by ahamdi — CHANGES_REQUESTED** [review]
This is a common component, it should be independent from applications usingit. Please pass the full search URL including the project ID

**Comment by hassenelkamel-exo** [comment]
> This is a common component, it should be independent from applications usingit. Please pass the full search URL including the project ID

File: webapp/portlet/src/main/webapp/common/js/SuggesterService.js [hunk 1]
```diff
@@ -38,11 +38,18 @@ function searchUsers(filter, items, typeOfRelation, searchOptions) {
...
+
+  let params = null;
+  let url = null;
+  if (searchOptions && !searchOptions.searchUrl) {
     Object.assign(options, searchOptions);
+    params = $.param(options);
+    url = '/portal/rest/social/people/suggest.json?'.concat(params);
+  } else {
+    url = searchOptions.searchUrl.concat(filter);
   }
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/common/js/SuggesterService.js:50]: Could we let it have other olptions like this ? WDYT ?

```suggestion
  if (searchOptions) {
    if(searchOptions.searchUrl) {
      url = searchOptions.searchUrl;
      delete searchOptions.searchUrl;
    } else {
      url = '/portal/rest/social/people/suggest.json?';
    }
    Object.assign(options, searchOptions);
    params = $.param(options);
    url = url.concat(params);
  }
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**